### ISSUE TITLE
Map is not displaying inherited non-public geolocations

### ISSUE THREAD

**Issue by txau** [huridocs/uwazidocs#4403]
When inheriting a geolocation property from a non-public entity, the map fails to render the markers if you don't have permissions.

Access control over inherited geolocations is supposed to be based on the *inheriting* entity. In fact, the entities display a map with the geolocation even if you are logged out. Nevertheless, the library map is not showing the markers for these inherited geolocations from a non-public entity.

This also happens if you are logged in with a user that does not have permissions for that information.

**Comment by mfacar** [comment]
@txau, I checked against master and there is the same issue there, it seems to be a pre-existing issue. I'll take a looka

**Comment by txau** [comment]
@LaszloKecskes I can confirm that this is a backend issue. The client request to `api/search` with the flag `geolocation=true` is returning an empty result list (that is, the rows array in the response contains zero items).

Tracking down the issue to the function `processGeolocationResults` in search.js. It takes as 3rd function argument `inheritedEntities` in order to grab the normalized value, instead of using the already present denormalized one at:

`inherit_geolocation: (inherited.metadata[targetProperty] || []).filter(p => p.value),`

When the entity from which it is inheriting is not accessible (due to permissions or sync), this value is empty and does not pass the test later on:
```
if (entityHasGeolocation(row)) {
      processedRows.push(row);
    }
```

Ideally inherited properties are fully denormalized and do not need to consult the donor entity when retrieving information. Rewriting this will not only fix the bug but also alleviate search.js and its complexity since several functions can be removed or reduced such as `searchGeolocations`, `getInheritedEntities` and `getInheritedEntitiesData`.

**Comment by LaszloKecskes** [comment]
@txau @mfacar I cannot replicate the issue. Please confirm that it still exists. If not, my best guess is that the root cause was the same as #4530, and fixed in [#4640](https://github.com/huridocs/uwazi/pull/4640).

**Comment by mfacar** [comment]
yes @LaszloKecskes, the issue is still present, you can replicate it when inheriting from a not public entity and check the map view without a logged user.
